diff.go

```diff
@@ -380,33 +391,95 @@ func diffComponentExpose(oldDef, newDef ComponentDefinition) []DiffInfo {
 
 	if oldExpose != newExpose {
 		diffInfos = append(diffInfos, DiffInfo{
-			Type: DiffInfoComponentExposeUpdated,
-			Old:  oldExpose,
-			New:  newExpose,
+			Type:      DiffTypeComponentExposeUpdated,
+			Component: componentName,
+			Old:       oldExpose,
+			New:       newExpose,
 		})
 	}
 
 	return diffInfos
 }
 
-func diffComponentScale(oldDef, newDef ComponentDefinition) []DiffInfo {
+// diffComponentScale checks in detail what diff type should be applied between
+// oldDef and newDef. The following can be applied.
+//   placement changed -> DiffTypeComponentScalePlacementUpdated
+//   min decreased     -> DiffTypeComponentScaleMinDecreased
+//   min increased     -> DiffTypeComponentScaleMinIncreased
+//   max decreased     -> DiffTypeComponentScaleMaxDecreased
+//   max increased     -> DiffTypeComponentScaleMaxIncreased
+func diffComponentScale(oldDef, newDef ComponentDefinition, componentName ComponentName) []DiffInfo {
+	oldScaleDef := &ScaleDefinition{}
```

Comment:
Not sure. This is just about having diffs. The code is doing that. What we would need to do is to default given definitions before creating the diff. As you said, we need to do this in a later step.